src/frontend/src/pages/MainPage.tsx

```diff
@@ -4,28 +4,143 @@ import Populate from '../components/mainpageleft/PopulateLeftSection';
-                  style={{ width: '1000px', float: 'right', height: '500px' }}
-               >
-                  <RightSection apt={to || apartments[0]} />
-               </div>
-               <div className="Left">
+         <Stack>
+            {/* Top bar */}
+            <AppBar component="nav">
+               <Toolbar>
+                  <Grid container spacing={0}>
+                     <Grid item>
+                        <Box>
+                           <Avatar>
+                              <AccountCircleIcon />
+                           </Avatar>
+                        </Box>
+                     </Grid>
+                     {logged === true && (
+                        <Grid item>
+                           <Box>
+                              <Button
+                                 sx={{ color: '#fff' }}
+                                 onClick={() => navigate('/user')}
+                              >
+                                 User
+                              </Button>
+                           </Box>
+                        </Grid>
+                     )}
+                     <Grid item xs={logged === true ? 10 : 11}>
+                        <Box>
+                           <Button sx={{ color: '#fff' }}>About</Button>
+                        </Box>
+                     </Grid>
+                     {logged === true && (
+                        <Grid item>
+                           <Box
+                              display="flex"
+                              justifyContent="flex-end"
+                              alignItems="flex-end"
+                           >
+                              <Button
+                                 sx={{ color: '#fff' }}
+                                 onClick={() => {
+                                    logout();
+                                    navigate('/login');
+                                 }}
+                              >
+                                 Log out
+                              </Button>
+                           </Box>
+                        </Grid>
+                     )}
+                     {logged === false && (
+                        <Grid item>
+                           <Box
+                              display="flex"
+                              justifyContent="flex-end"
+                              alignItems="flex-end"
+                           >
+                              <Button
+                                 sx={{ color: '#fff' }}
+                                 onClick={() => {
+                                    navigate('/login');
+                                 }}
+                              >
+                                 Log in
+                              </Button>
+                           </Box>
+                        </Grid>
+                     )}
+                  </Grid>
+               </Toolbar>
+            </AppBar>
```

Comment:
If we're logged out, shouldn't we also have a `sign up` button

Reply:
We already have sign up from log in.

Reply:
yeah, but a user might think they have an account

Reply:
I think usually websites cache your session so that it displays `sign up` if you don't have account or `log in` if it remembers